### ISSUE TITLE
Libraries: non-admin users with `manage` permission cannot really manage.

### ISSUE THREAD

**Issue by davelopez** [galaxyproject/galaxy#12003]
I may not have understood well how non-admin users are supposed to manage libraries but here is a use case I found strange:

An Admin changes the permissions of a library to the following:

![image](https://user-images.githubusercontent.com/46503462/118798694-36e12380-b89e-11eb-9ed7-437a5398c3aa.png)

Then the `t01@test.gx` user logs in and tries to manage the permissions of that library:

- If the user tries to save the permissions without changes, an error is shown. With the response error being:
  ````JSON
  {"err_msg": "No accessible role found with the id provided.", "err_code": 400008}
  ````
- If the user tries to add another role, only the user's own roles can be selected. This may depend on the `expose_user_email` config option or any other configuration but renders the selector unusable in this case.
  ![image](https://user-images.githubusercontent.com/46503462/118799645-40b75680-b89f-11eb-83c4-a58a855db550.png)

In summary, the user can only save the permissions if she/he has access to all the roles.

Is this the intended behavior?

**Comment by mvdbeek** [comment]
@davelopez do you know if this is still a problem after #11701 ?

**Comment by mvdbeek** [comment]
Hmm, I think all of this should not require being an admin, that might have just been a shortcut to make the UI less confusing and adapt it to the current broken state.

I think those roles are not hierarchical, so 
> But the user t02@test.gx can effectively add items to the library while the others can't (so this one it's working).

seems correct ?

**Comment by davelopez** [comment]
> Hmm, I think all of this should not require being an admin, that might have just been a shortcut to make the UI less confusing and adapt it to the current broken state.

Makes sense, otherwise, I see not much point in having those different roles.

> seems correct?

Yeah, the `add items` roles seem to be working fine, they show the add dataset, etc. buttons only if the user has the correct role. 
I think we need to show/enable the correct buttons here depending on the `manage` and `edit` roles and then check if there is something wrong in the backend when trying to manage roles with a non-admin user.
